server/src/main/java/org/elasticsearch/indices/NodeIndicesStats.java

```diff
@@ -66,10 +71,21 @@ public NodeIndicesStats(StreamInput in) throws IOException {
             }
             statsByShard.put(index, indexShardStats);
         }
+
+        statsByIndex = new HashMap<>();
+        if (in.getVersion().onOrAfter(VERSION_SUPPORTING_STATS_BY_INDEX)) {
+            entries = in.readVInt();
```

Comment:
We can just use `in.readMap(Index::new, CommonStats::new)` here can't we?

Reply:
Good idea!